core/contracts/composed_stream_template.kf

```diff
@@ -795,10 +795,21 @@ procedure set_taxonomy($data_providers text[], $stream_ids text[], $weights deci
     if $length != array_length($weights) {
         error('data_providers and weights must have the same length');
     }
+    if $length != array_length($start_dates) {
+        error('data_providers and start_dates must have the same length');
+    }
 
     for $i in 1..$length {
-        INSERT INTO taxonomies (child_stream_id, child_data_provider, weight, created_at, version)
-            VALUES ($stream_ids[$i], $data_providers[$i], $weights[$i], $block_height, $next_version);
+        $current_uuid :=  uuid_generate_v5($current_uuid, @txid);
+
+        // if start_date is empty, we set it to a default value
```

Comment:
if null start date has a meaning (fallback) wouldn't be better to let it be null? and empty strings coalesced to null